src/cache/mod.rs

```diff
@@ -1,4 +1,4 @@
-mod memory;
+pub mod memory;
```

Comment:
Why this is needed?

Reply:
the compiler complained because it's a private mod when I started to use the `set` and `get` MemCache methods.

Reply:
because u imported it as `use crate::cache::memory::MemCache;` not as `use crate::cache::MemCache;` 
You already can see 2 lines down it's exposed with the `pub use memory::MemCache;`

Reply:
got it